go/kex2/provisioner.go

```diff
@@ -149,6 +154,9 @@ func (p *provisioner) runProtocolWithCancel() (err error) {
 		p.canceled = true
 		return ErrCanceled
 	case err = <-ch:
+		if _, ok := err.(rpc.CanceledError); ok && !p.helloReceived {
+			return ErrHelloTimeout
```

Comment:
That's not the error that goes over RPC.  The engine checks for it and returns its own error.  But the engine error isn't exportable, so I'll fix that.